packages/chain-substrate/src/implementation.ts

```diff
@@ -71,39 +71,43 @@ export class SubstrateChainImplementation implements ChainImplementation<Extensi
 	async signSession({ extension, address }: ExtensionAndAddress, payload: SessionPayload): Promise<Session> {
 		if (extension.signer.signRaw === undefined) throw new Error("Invalid signer")
 		const message = stringToHex(getSessionSignatureData(payload))
-		const signature = (await extension.signer.signRaw({
-			address,
-			data: message,
-			type: 'bytes',
-		})).signature;
+		const signature = (
+			await extension.signer.signRaw({
+				address,
+				data: message,
+				type: "bytes",
+			})
+		).signature
 		const session: Session = { type: "session", payload, signature }
 		return session
 	}
 
 	async signAction({ extension, address }: ExtensionAndAddress, payload: ActionPayload): Promise<Action> {
 		if (extension.signer.signRaw === undefined) throw new Error("Invalid signer")
 		const message = stringToHex(getActionSignatureData(payload))
-		const signature = (await extension.signer.signRaw({
-			address,
-			data: message,
-			type: 'bytes',
-		})).signature;
+		const signature = (
+			await extension.signer.signRaw({
+				address,
+				data: message,
+				type: "bytes",
+			})
+		).signature
 		const action: Action = { type: "action", payload, session: null, signature }
 		return action
 	}
 
 	async signDelegatedAction(mnemonic: SubstrateMnemonic, payload: ActionPayload) {
-		const keyring: Keyring = new Keyring({ ss58Format: 42 });
-		const pair = keyring.addFromUri(mnemonic, {}); // use sr25519 by default
+		const keyring: Keyring = new Keyring({ ss58Format: 42 })
+		const pair = keyring.addFromUri(mnemonic, {}) // use sr25519 by default
 		const message = getActionSignatureData(payload)
-		const signatureBytes = pair.sign(message);
-		const signature = new Buffer(signatureBytes).toString('hex');
+		const signatureBytes = pair.sign(message)
+		const signature = Buffer.from(signatureBytes).toString("hex")
 
 		const action: Action = {
 			type: "action",
 			payload: payload,
 			session: pair.address,
-			signature,
+			signature: `0x${signature}`,
```

Comment:
Do we want to return this? The verify function only works if the signature is prefixed with `0x`.

Reply:
if that's what verify expects then definitely yes